syntax = "proto3";

package mdlidar;

option cc_enable_arenas = true;

message Vector3D {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Orientation {
  Vector3D axis = 1;
  float angle = 2;
}

message State {
  Vector3D position = 1;
  Orientation orientation = 2;
  Vector3D position_p = 3;
  Orientation orientation_p = 4;
}

message Point {
  float range = 1;
  float rotation = 2;
  int32 object_id = 3;
  bool dynamic = 4;
}

message Frame {
  int64 start_time = 1;
  int64 end_time = 2;
  repeated Point points = 3;
  State state = 4;
  repeated State object_state = 5;
}

message Header {
  repeated uint32 object_ids = 1;
  float delta_time = 2;
  int32 channels_count = 3;
  uint32 points_count_by_channel = 4;
}